plugin/trino-hive/src/main/java/io/trino/plugin/hive/s3/TrinoS3FileSystem.java

```diff
@@ -938,10 +942,27 @@ private AWSCredentialsProvider createAwsCredentialsProvider(URI uri, Configurati
                 .orElseGet(DefaultAWSCredentialsProviderChain::getInstance);
 
         if (iamRole != null) {
-            provider = new STSAssumeRoleSessionCredentialsProvider.Builder(iamRole, "trino-session")
-                    .withExternalId(externalId)
-                    .withLongLivedCredentialsProvider(provider)
-                    .build();
+            String stsEndpointOverwrite = conf.get(STS_ENDPOINT);
+            String stsRegionOverwrite = conf.get(STS_REGION);
+
+            if (stsEndpointOverwrite != null && stsRegionOverwrite != null) {
+                AWSSecurityTokenServiceClientBuilder stsClientBuilder =
+                        AWSSecurityTokenServiceClientBuilder.standard()
+                        .withCredentials(provider);
+                stsClientBuilder.withEndpointConfiguration(
```

Comment:
If `stsRegionOverwrite` is set but `stsEndpointOverwrite` is not it looks like we want to call `stsClientBuilder.withRegion`

Reply:
Good point, I will update that